gym/envs/registration.py

```diff
@@ -89,7 +89,15 @@ def spec(self, id):
         try:
             return self.env_specs[id]
         except KeyError:
-            raise error.UnregisteredEnv('No registered env with id: {}'.format(id))
+            # Parse the env name and check to see if it matches the non-version
+            # part of a valid env (could also check the exact number here)
+            env_name = match.group(1)
+            matching_envs = [full_env_name for full_env_name in self.env_specs
+                             if full_env_name.startswith(env_name + '-v')]
```

Comment:
I would prefer to do the env name parsing in Spec, and then the comparison becomes a straightforward equality comparison (and the semantics are dictated entirely by the regex).